yesod-bin/AddHandler.hs

```diff
@@ -214,3 +221,10 @@ mkHandler name pattern methods = unlines
 uncapitalize :: String -> String
 uncapitalize (x:xs) = toLower x : xs
 uncapitalize "" = ""
+
+getSrcDir :: FilePath -> IO FilePath
+getSrcDir cabal = do
+    pd <- flattenPackageDescription <$> readPackageDescription normal cabal
+    let buildInfo = allBuildInfo pd
+        srcDirs = concatMap hsSourceDirs buildInfo
+    return $ head srcDirs
```

Comment:
but it looks already having default value  with current directory for `hs-source-dirs`.
https://github.com/haskell/cabal/blob/master/Cabal/Distribution/PackageDescription/Configuration.hs#L580-L584